src/encoded/static/libs/ui/node_graph.js

```diff
@@ -0,0 +1,254 @@
+            if (d.depth === 0) {
+                d.y = d.depth * 100;
+            } else {
+                d.y = (d.depth * height) / maxNodeDepth;
+            }
+        });
+
+        const node = svg.selectAll('g.node')
+            .data(nodes, (d) => d.id);
+
+        const nodeEnter = node.enter().append('g')
+            .attr('class', (d) => `node ${d.data.class} ${d.data.name}`)
+            .attr('transform', `translate(${source.x0},${source.y0})`)
+            .on('click', (e, d) => {
+                setSelectedNodes(d.data.name);
+                const clickedName = nodeKeyName(d.data.name);
+                if (internalSelectedNodes.indexOf(clickedName) > -1 && internalSelectedNodes.length > 1) {
+                    internalSelectedNodes = internalSelectedNodes.filter((s) => s !== clickedName);
+                } else if (internalSelectedNodes.indexOf(clickedName) > -1) {
+                    internalSelectedNodes = [];
+                } else {
+                    internalSelectedNodes = [...internalSelectedNodes, clickedName];
+                }
+                const clickedCell = d3.select(`.js-cell-${nodeKeyName(d.data.name)}`);
+                clickedCell.attr('class', `${internalSelectedNodes.indexOf(clickedName) > -1 ? 'active-cell' : ''} js-cell-${clickedName} js-cell ${d._children ? 'parent-cell' : ''}`);
+                e.stopPropagation();
+            });
+
+        const nodeGroup = nodeEnter.append('g')
+            .attr('class', (d) => `js-cell-${nodeKeyName(d.data.name)} js-cell ${internalSelectedNodes.indexOf(nodeKeyName(d.data.name)) > -1 ? 'active-cell' : ''} ${d._children ? 'parent-cell' : ''}`)
+            .on('mouseover', (e, d) => {
+                // add hover styles on node
+                d3.select(`.js-cell-${nodeKeyName(d.data.name)}`).selectAll('ellipse').style('transform', 'scale(1.2)');
+                d3.select(`.js-cell-${nodeKeyName(d.data.name)}`).selectAll('ellipse').attr('class', `hover-class ${d.data.class ? d.data.class : 'default'}`);
+                // highlight corresponding matrix row
+                d3.select(`.${d.data.name.replace(/\s/g, '_').toLowerCase()}`).selectAll('th').attr('class', d.data.class);
+            })
+            .on('mouseout', (e, d) => {
+                // remove hover styles from node
+                d3.select(`.js-cell-${nodeKeyName(d.data.name)}`).selectAll('ellipse').style('transform', 'scale(1)');
+                d3.select(`.js-cell-${nodeKeyName(d.data.name)}`).selectAll('ellipse').attr('class', (d.data.class ? d.data.class : 'default'));
+                // remove highlight from corresponding matrix row
+                d3.select(`.${d.data.name.replace(/\s/g, '_').toLowerCase()}`).selectAll('th').attr('class', '');
+            });
+
+        ellipseSettings.forEach((ellipseSetting) => {
+            nodeGroup.append('ellipse')
+                .attr('cx', ellipseSetting.cx)
+                .attr('cy', ellipseSetting.cy)
```

Comment:
This code is feel. I just got a feeling it can be done with JSON/object rather than adding .attr one at a time.